src/mondefense.cpp

```diff
@@ -1,38 +1,47 @@
+#include "mondefense.h"
 #include "monster.h"
+#include "creature.h"
+#include "damage.h"
 #include "game.h"
 #include "rng.h"
 #include "line.h"
 #include "bodypart.h"
-#include "speech.h"
 #include "messages.h"
-#include "mondefense.h"
-#include <algorithm>
 
-
-void mdefense::zapback(monster *m, Creature *source, const projectile *proj)
+void mdefense::zapback(monster *const m, Creature *const source, projectile const* const proj)
 {
-    player *foe = dynamic_cast< player* >( source );
-    monster *othermon = dynamic_cast< monster* >( source );
-    if( rl_dist( m->pos(), source->pos() ) > 1 ) {
-        return; // Out of range
+    if ((proj) ||                              // Not a melee attack
+        (rng(0, 100) > m->def_chance) ||       // Attacker lucked out
+        (rl_dist(m->pos(), source->pos()) > 1) // Out of range
+    ) {
+        return;
     }
-    if( proj != nullptr || rng(0, 100) > m->def_chance ) {
-        return; // Not a melee attack or attacker lucked out
+
+    auto const is_eligible_player = [](player const *const foe) {
```

Comment:
Yes, KISS is a fine principal to always keep in mind. I agree. I also think one should try to do the "right thing" if it's not overly burdensome to do so. Here those variable ought to be const, no? Here the code ought to use short circuited evaluation, no?

I know I've seen a better article somewhere (I think maybe linked from isocpp.org?), but these illustrate the "why lambda?"
[Stackoverflow](http://stackoverflow.com/questions/21450994/complex-initialization-for-a-const-variable-while-avoiding-new-allocation).
[Sutter's mill](http://herbsutter.com/2013/04/05/complex-initialization-for-a-const-variable/).